phylogenetic/defaults/config_dengue.yaml

```diff
@@ -15,7 +15,7 @@ filter:
     genome: 5000
     E: 1000
   sequences_per_group:
-    all: '10'
+    all: '36'
```

Comment:
Are you still considering replacing `sequences_per_group` with `subsample_max_sequence` as described in #90?

> Instead of backwards calculating the `all` `sequences_per_group` to get ~4000 samples, it was suggested that we could **set `subsample_max_sequence` to `4000`**.

I think this would be more intuitive.

Reply:
Thanks for asking! I should have documented that using `subsample_max_sequences` (and omitting the group_by year option) results in the loss of temporal signal needed to infer a molecular clock rate. ([See the x-axis for the tangletree on the left as compared to the original on the right](https://next.nextstrain.org/staging/dengue/trials/maxsequences/dengue/all/genome:dengue/all/genome)).

* https://github.com/nextstrain/dengue/issues/90#issuecomment-2573595396

Reply:
The loss of temporal signal should only be due to removing `year` from `group_by`.

It should still make sense to replace `sequences_per_group` with `subsample_max_sequence`, since what `augur filter` [does internally](https://github.com/nextstrain/augur/blob/1843a9b02d5380e81d4adb874db85704985670b4/augur/filter/subsample.py#L577) is calculate a value for `sequences_per_group` based on `subsample_max_sequence` + group sizes.

Reply:
Ohh, thanks for pointing this out! I should have red the documentation more careful for `subsample-max-sequences`. 

* Implemented in https://github.com/nextstrain/dengue/pull/91/commits/0d73b74fa524775c3207fff886b91a424a7a8df2 
* And the trees looks reasonable as far as I can tell
    * https://next.nextstrain.org/staging/dengue/trials/subsamplemaxseq2025/dengue/all/genome:dengue/all/genome
    * https://next.nextstrain.org/staging/dengue/trials/subsamplemaxseq2025/dengue/all/E:dengue/all/E

Reply:
Ah, perhaps I posted too soon, I'm glancing at the E tree...

Reply:
Looks like genome and E trees were fixed by running it again. I'm good with the approach of using the combination of `--subsample-max-sequences --group-by`, thanks! This PR is ready for review